release-notes/cabal-install-3.16.0.0.md

```diff
@@ -0,0 +1,170 @@
+  - `haddock` receives `--package-name` flag whcih allows to set names of
+    components which are included in the main `index.html` file.
+  - added `--use-unicode` flag to `haddock` and `haddock-project` commands.
+  - The directory structure of `./dist-newstyle` has changed.  `haddock`
+    subcommand will install `package:sublib` component in a directory
+    `package/sublib` under `l/sublib/doc/html/`.  This is important for
+    `haddock-project` command and in the future might will be useful for hackage
+    support of sublibraries.  See
+    https://github.com/haskell/cabal/pull/9821#discussion_r1548557115.
+
+### Other changes
+
+- Add support for building profiled dynamic way [#4816](https://github.com/haskell/cabal/issues/4816) [#9900](https://github.com/haskell/cabal/pull/9900)
+
+
+  New options for `cabal.project` and `./Setup` interface:
+
+  * `profiling-shared`: Enable building profiling dynamic way
+  * Passing `--enable-profiling` and `--enable-executable-dynamic` builds
+    profiled dynamic executables.
+
+  Support for using `profiling-shared` is guarded behind a constraint
+  which ensures you are using `Cabal >= 3.13`.
+
+  In the cabal file:
+
+  * `ghc-prof-shared-options`, for passing options when building in
+    profiling dynamic way
+
+- Fix interaction of `--*-shared` and `--*-executable-dynamic` options. [#10050](https://github.com/haskell/cabal/issues/10050) [#9900](https://github.com/haskell/cabal/pull/9900)
+
+  If you explicitly request `--disable-shared` it should disable the building of
+  a shared library and override any automatic ways this option is turned on.
+
+  Passing `--enable-executable-dynamic` turns on `--enable-shared` if the option is
+  not specified explicitly.
+
+  Before this patch, writing `--disable-shared` on its own would not disable the building of shared libraries. Writing `--disable-shared` and `--disable-executable-dynamic` would disable shared library
+  creation (despite `--disable-executable-dynamic` being the default).
+
+  Now:
+
+  * If you specify `--enable-shared` then shared objects are built.
+  * If you specify `--disabled-shared` then shared objects are not built.
+  * If you don't explicitly specify whether you want to build shared libraries then
+    * `--enable-executable-dynamic` will automatically turn on building shared libraries
+    * `--enable-executable-dynamic --enable-profiling` will automatically turn on building
+      shared profiling libraries (if supported by your compiler).
+
+- Working directory support for `Cabal` [#9702](https://github.com/haskell/cabal/issues/9702) [#9718](https://github.com/haskell/cabal/pull/9718)
+
+  The `Cabal` library is now able to handle a passed-in working directory, instead
+  of always relying on the current working directory of the parent process.
+
+  In order to achieve this, the `SymbolicPath` abstraction was fleshed out, and
+  all fields of `PackageDescription` that, if relative, should be interpreted
```

Comment:
This is present in the Cabal changelog too, right? Having Cabal-the-library API description in cabal-the-tool changelog seems off.

Reply:
I understand this should have been done earlier (at PR time),  but from a `cabal-install` user, what does change?

Reply:
It should be invisible to users, but any time `cabal-install` calls into `Cabal` it has to use the new path structure instead of `FilePath`s. So every file that uses functions from `Cabal` changed.

Possibly this means that it shouldn't have been listed for `cabal-install` since a user-visible change would be a bug.

Reply:
> It should be invisible to users

Fantastic, then imho it should go from here.